### ISSUE TITLE
make dggridr easier to use

### ISSUE THREAD

**Issue by 7yl4r** [marinebon/obisindicators#31]
Several of us have had issues getting `dggridr` to work properly. The DESCRIPTION file currently has this set up to build it from C source code. We could try taking a different approach.

From Pieter:

>  I also had some issues with dggridr on a new machine, and I ended up using st_make_grid() with an equal area projection (EPSG:3410)

We could also try using [the H3 grid](https://eng.uber.com/h3/).

**Comment by 7yl4r** [comment]
Revisiting this. Here are ways forward I see:

1. work with the dggridr maintainer to make the CRAN version work for us
2. replace dggridr usage here with
    1. h3 grid
    2. st_make_grid 

2.2 is probably easiest. 1 would be a good service for the community in general. 2.1 is the most forward-looking.

**Comment by 7yl4r** [comment]
Yeah, there is some javascript in the app that does something related to the dateline issue but obviously it isn't perfect. 

For obisindicators we don't need to display the hexagons though, I just want to calculate es50 for each h3 hex.

**Comment by 7yl4r** [comment]
Working to use H3 on the [H3 branch](https://github.com/marinebon/obisindicators/tree/h3).

**Comment by bbest** [comment]
Next step: split anti-meridonal hexagons into two with same h3 ID and still simple `POLYGON` not `MULTIPOLYGON`, as is done by [`sf::st_wrap_dateline()`](https://r-spatial.github.io/sf/reference/st_transform.html)

**Comment by MathewBiddle** [comment]
Please please please remove the dependency on `dggridr`. The installation process is a nightmare with that package.

**Comment by 7yl4r** [comment]
I feel like there is some projection setting that will handle this hexagon wrapping issue that we are having on the h3 branch. Some discussion on [this issue](https://github.com/r-spatial/sf/issues/280) seems relevant. I am going to try playing with some of these sf features tomorrow.

**Comment by 7yl4r** [comment]
It works!!!

I don't want to close this quite yet though because the builds are failing to build from the DESCRIPTION.

>   * local::.: Can't install dependency h3
>  * h3: Can't find package called h3. 

The remote is pointing to the correct repo though... so I don't know what the problem is.

**Comment by MathewBiddle** [comment]
I'll give this a try sometime soon 🤞

**Comment by bbest** [comment]
I wonder if the issue here has to do with the mismatch between the repo name `h3-r` and the package name  `h3`:

https://github.com/marinebon/obisindicators/blob/06e5d2fafed3f974f99b1757126d7b467d17c496/DESCRIPTION#L34-L45

**Comment by 7yl4r** [comment]
> I wonder if the issue here has to do with the mismatch between the repo name `h3-r` and the package name `h3`:
> 
> https://github.com/marinebon/obisindicators/blob/06e5d2fafed3f974f99b1757126d7b467d17c496/DESCRIPTION#L34-L45

That's the best theory so far. We may need to report this to the github action.

I tried another syntax for the DESCRIPTION file and now I am getting a different error:

```
 Error: Error: <callr_remote_error: Cannot install packages:
  * deps::.: Can't install dependency github::crazycapivara/h3-r
  * pkgdown: dependencies must be TRUE, FALSE, NA or a list of dependency types>
```

**Comment by MathewBiddle** [comment]
h3 grid calculations take forever. I wonder if `h3o` is a package to explore to speed up the process?

https://josiahparry.com/projects/pkgs/h3o.html

I don't think it would change a lot of the obisindicators package...
